src/formats.ts

```diff
@@ -65,6 +65,18 @@ export function isSupportedLocale(locale: string): locale is SupportedLocale {
   return Object.values(SupportedLocale).indexOf(locale) !== -1;
 }
 
+export function getSafeLocaleFormat(locale) {
+  if (!isSupportedLocale(locale)) {
+    const [countryCode] = locale.split('-');
+    if (isSupportedLocale(countryCode)) {
+      return countryCode;
+    }
+    return null;
+  }
+
+  return locale;
```

Comment:
I think we can flatten this out like so:

```typescript
if (isSupportedLocale(locale)) {
  return locale;
}

const [countryCode] = locale.split('-');

if (isSupportedLocale(countryCode)) {
  return countryCode;
}

return null;
```

Double check that it still works, but I think it reads a little easier :)

Reply:
Yes! This is what I was looking for but didn't get there. 😆

Reply:
Tests pass!